### PR TITLE
Binary read support

### LINKED ISSUE #725: [bug] None of the assets functions except assets.copy() implement the remote API
**Describe the bug**
When I call the eldritch function assets.read(), I would expect it to implement the remote API where I could read something off the remote server into a string that can be used on the agent. It appears that only assets.copy() implements the remote API. 

**To Reproduce**

Write a tome using anything in the assets library that is not assets.copy() and attempt to retrieve an asset from tavern. 

**Expected behavior**
The expected behavior would be for all of the functions in the assets library to implement the remote API. 

**Screenshots**
image

**Additional context**
The screenshot above demonstrates an attempt at using assets.read() to read in the content of a file as a string, using an asset that was staged in [name_of_tome]/assets/[ASSET]

### PR DESCRIPTION
**Pull request by hulto**
#### What type of PR is this?
/kind bug
/kind feature

#### What this PR does / why we need it:
This PR adds support for reading binary files and adds remote support to the remaining asset functions:
- read
- read_binary
- list

#### Which issue(s) this PR fixes:
Fixes #725

### CODE REVIEW

File: implants/lib/eldritch/src/assets/read_binary_impl.rs [hunk 1]
```diff
@@ -1,6 +1,31 @@
...
-pub fn read_binary(src: String) -> Result<Vec<u32>> {
+use anyhow::{Context, Result};
+use pb::c2::FetchAssetResponse;
+use starlark::{eval::Evaluator, values::list::ListRef};
+
+use crate::runtime::{messages::FetchAssetMessage, Environment};
+
+fn read_binary_remote(rx: Receiver<FetchAssetResponse>) -> Result<Vec<u32>> {
+    // Wait for our first chunk
+    let resp = rx.recv()?;
```
Comment (Cictrone) [inline implants/lib/eldritch/src/assets/read_binary_impl.rs:11]: is there a reason we do the first chunk separately?

File: implants/lib/eldritch/src/assets/read_binary_impl.rs [hunk 2]
```diff
@@ -1,6 +1,31 @@
...
+    for resp in rx {
+        let mut new_chunk = resp.chunk.iter().map(|x| *x as u32).collect::<Vec<u32>>();
+
+        res.append(&mut new_chunk);
+    }
+
+    Ok(res)
+}
+
+fn read_binary_local(src: String) -> Result<Vec<u32>> {
```
Comment (Cictrone) [inline implants/lib/eldritch/src/assets/read_binary_impl.rs:28]: i think maybe embedded might be less ambiguous than local? That being said, im sure all the file functions needs some sort of adjustment lol.

File: implants/lib/eldritch/src/assets/read_binary_impl.rs [hunk 3]
```diff
@@ -41,4 +86,91 @@ mod tests {
...
+                continue;
+            }
+
+            // Ensure the right asset was requested
+            assert!(fetch_asset_msgs.len() == 1);
+            let msg = fetch_asset_msgs.pop().expect("no asset request received!");
+            assert!(msg.name == "remote_asset/just_a_remote_asset.txt");
+
+            // Now, we provide the file to eldritch (as a series of chunks)
+            msg.tx
```
Comment (Cictrone) [inline implants/lib/eldritch/src/assets/read_binary_impl.rs:136]: eventually we will setup mock testing to make this even nicer but this looks nice for now!

File: implants/lib/eldritch/src/assets/read_impl.rs [hunk 4]
```diff
@@ -12,14 +18,171 @@ pub fn read(src: String) -> Result<String> {
...
+
 #[cfg(test)]
 mod tests {
-    use super::*;
+    use std::collections::HashMap;
+
+    use crate::runtime::{messages::FetchAssetMessage, Message};
+    use pb::{c2::FetchAssetResponse, eldritch::Tome};
+
+    // #[test]
```
Comment (Cictrone) [inline implants/lib/eldritch/src/assets/read_impl.rs:62]: pls delete commented code

File: implants/lib/eldritch/src/file/template_impl.rs [hunk 5]
```diff
@@ -21,8 +21,10 @@ pub fn template(
     autoescape: bool,
 ) -> Result<()> {
     let context = build_context(args)?;
-    let template_content = fs::read_to_string(template_path)?;
-    let res_content = Tera::one_off(template_content.as_str(), &context, autoescape)?;
+    let data = fs::read(template_path.clone())?;
```
Comment (Cictrone) [inline implants/lib/eldritch/src/file/template_impl.rs:24]: whats the difference here?

**Review by Cictrone — APPROVED** [review]
also pls make sure there are no lints

File: implants/lib/eldritch/src/file/template_impl.rs (hunk 5 above)
Comment (hulto) [reply implants/lib/eldritch/src/file/template_impl.rs:24]: Won't fail when reading non-UTF8 characters like a binary file.

Comment (Cictrone) [reply implants/lib/eldritch/src/file/template_impl.rs:24]: ah, i see.

File: implants/lib/eldritch/src/assets/read_binary_impl.rs (hunk 1 above)
Comment (hulto) [reply implants/lib/eldritch/src/assets/read_binary_impl.rs:11]: I don't think so tests pass without it.

**Comment by hulto** [comment]
No lints
image

**Review by Cictrone — APPROVED** [review]
perfecto

**Pull request merged**